Framework/NexusGeometry/src/NexusGeometryParser.cpp

```diff
@@ -595,6 +602,109 @@ void parseMonitors(const H5::Group &root, InstrumentBuilder &builder) {
   }
 }
 
+class Tube {
```

Comment:
`NexusGeometryParser` is big enough, so move to another file.

Reply:
Description of tube required. 

i.e. Supports intermediate calculation. Mutable structure built to house detectors that form part of the tube.